mlflow/pyspark/ml/__init__.py

```diff
@@ -1077,6 +1087,27 @@ def _infer_model_signature(input_example_slice):
             else:
                 _logger.warning(_get_warning_msg_for_skip_log_model(spark_model))
 
+        if log_datasets:
+            try:
+                # create a CodeDatasetSource
+                context_tags = context_registry.resolve_tags()
+                source = CodeDatasetSource(
+                    mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                    mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                )
```

Comment:
SparkML already logs the SparkDatasource. I think we should use that one instead. 
https://github.com/mlflow/mlflow/blob/master/mlflow/spark.py#L937